x/supplier/keeper/claim.go

```diff
@@ -9,19 +9,20 @@ import (
 	"github.com/pokt-network/poktroll/x/supplier/types"
 )
 
-// InsertClaim adds a claim to the store
-func (k Keeper) InsertClaim(ctx sdk.Context, claim types.Claim) {
-	logger := k.Logger(ctx).With("method", "InsertClaim")
+// UpsertClaim inserts or updates a specific claim in the store by index.
+func (k Keeper) UpsertClaim(ctx sdk.Context, claim types.Claim) {
+	logger := k.Logger(ctx).With("method", "UpsertClaim")
 
 	claimBz := k.cdc.MustMarshal(&claim)
 	parentStore := ctx.KVStore(k.storeKey)
 
 	// Update the primary store: ClaimPrimaryKey -> ClaimObject
 	primaryStore := prefix.NewStore(parentStore, types.KeyPrefix(types.ClaimPrimaryKeyPrefix))
-	primaryKey := types.ClaimPrimaryKey(claim.SessionId, claim.SupplierAddress)
+	sessionId := claim.GetSessionHeader().GetSessionId()
+	primaryKey := types.ClaimPrimaryKey(sessionId, claim.SupplierAddress)
 	primaryStore.Set(primaryKey, claimBz)
 
-	logger.Info("inserted claim for supplier %s with primaryKey %s", claim.SupplierAddress, primaryKey)
+	logger.Info("upserted claim for supplier %s with primaryKey %s", claim.SupplierAddress, primaryKey)
```

Comment:
```suggestion
	logger.Info(fmt.Sprintf("upserted claim for supplier %s with primaryKey %s", claim.SupplierAddress, primaryKey))
```
Same for any other logs